dtests/src/test/scala/io/snappydata/hydra/cdcConnector/DmlOpsInBatchCdcConnectorTest.scala

```diff
@@ -0,0 +1,714 @@
+            }
+        }
+
+        catch {
+            case e: FileNotFoundException => {
+            }
+            case io: IOException => {
+            }
+        }
+        queryList
+    }
+
+    def executeQueries(queryArray: ArrayList[String], conn: Connection) {
+        try {
+            val statement = conn.createStatement()
+            for (i <- 0 to queryArray.size()) {
+                var query = queryArray.get(i)
+                statement.executeUpdate(query)
+            }
+        }
+        catch {
+            case e: Exception => {
+                System.out.println("Caught exception " + e.getMessage)
+            }
+        }
+    }
+
+    def getANetConnection(netPort: Int): Connection = {
+        val driver = "io.snappydata.jdbc.ClientDriver"
+        Utils.classForName(driver).newInstance
+        DriverManager.getConnection("jdbc:snappydata:thrift://localhost[1527]")
+    }
+
+    def collectData(queryPath: String, query1: String,
+        snappyQuery: String, sqlQuery: String): Unit = {
+
+        println("Execute queries and print results")
+        val qArr = getQuery(queryPath)
+        val conn = getSqlServerConnection
+        val snappyconn = getANetConnection(1527)
+        executeQueries(qArr, conn)
+        Thread.sleep(15000)
+        var rs = snappyconn.createStatement(
+            ResultSet.TYPE_SCROLL_INSENSITIVE, ResultSet.CONCUR_READ_ONLY).executeQuery(query1)
+        while(!rs.next()){
+            rs = snappyconn.createStatement(
+                ResultSet.TYPE_SCROLL_INSENSITIVE, ResultSet.CONCUR_READ_ONLY).
+                executeQuery(query1)
+        }
+        val snappyDF = snappyconn.createStatement(
+            ResultSet.TYPE_SCROLL_INSENSITIVE, ResultSet.CONCUR_READ_ONLY)
+            .executeQuery(snappyQuery)
+        val sqlResultSet = conn.createStatement(
+            ResultSet.TYPE_SCROLL_INSENSITIVE, ResultSet.CONCUR_READ_ONLY)
+            .executeQuery(sqlQuery)
+        performValidation(sqlResultSet, snappyDF)
+
+    }
+
+    def performValidation(sqlResultSet: ResultSet, snappyResultSet: ResultSet): Unit = {
+
+        val resultMetaData = sqlResultSet.getMetaData
+        val columnCnt = resultMetaData.getColumnCount
+        val snappyMetadata = snappyResultSet.getMetaData
+        val snappyColCnt = snappyMetadata.getColumnCount
+        var sqlRowsCnt = 0
+        while(sqlResultSet.next()){
+            sqlRowsCnt = sqlRowsCnt + 1
+        }
+        var snappyRowsCnt = 0
+        while(snappyResultSet.next()){
+            snappyRowsCnt = snappyRowsCnt + 1
+        }
+        sqlResultSet.beforeFirst()
+        snappyResultSet.beforeFirst()
+        println("Row cnt of snappy and sql table is = " + snappyRowsCnt + " " + sqlRowsCnt)
+        println("Column cnt of snappy and sql table is = " + snappyColCnt + " " + columnCnt)
+        if (snappyRowsCnt.equals(sqlRowsCnt)) {
+            while (sqlResultSet.next()) {
+                snappyResultSet.next()
```

Comment:
How do we know if the results are in same order? Are there only one row as result?

Reply:
For some test cases there is only one row in resultset and for some test cases if there are multiple rows then selecting records in order.